kong/core/handler.lua

```diff
@@ -79,6 +79,8 @@ return {
 
       ctx.KONG_ACCESS_START = get_now()
 
+      local original_host_header = ngx.req.get_headers().host
```

Comment:
all the logic requires the hostname only, so an extra field would have to be returned, the original host-header. I see what you mean, but don't see a clean solution.